includes/gateway/class-omise-payment-fbbot.php

```diff
@@ -0,0 +1,224 @@
+			global $wp_query;
+
+			if ( ! isset( $wp_query->query_vars['error_message'] ) ) {
+			  return '<strong>Woocommerce system has error. Please try again.</strong>';
+			}
+
+			$error_message = $wp_query->query_vars['error_message'];
+
+			return '<strong>' . $error_message . '</strong>';
+		}
+
+		public function process_payment_by_bot( $params, $order ) {
+			$charge_params = array(
+				'amount'      => $this->format_amount_subunit( $order->get_total(), $order->get_currency() ),
+				'currency'    => $order->get_currency(),
+				'description' => 'OrderID is '.$order->get_order_number().' : This order created from Omise FBBot and CustomerID is ' . $params['messenger_id'],
+				'metadata' => $params['metadata'],
+				'card' => $params['omise_token']
+		    );
+
+		    if ( $this->omise_3ds ) {
+		    	$return_uri =  site_url() . '/complete-payment';
+
+		      	$charge_params['return_uri'] = $return_uri;
+		    }
+
+		    // Create Charge
+		    try {
+		      	$charge = OmiseCharge::create( $charge_params, '', $this->secret_key() );
+		      	// We move checking charge status to request handler in handle triggered from omise method
+
+		      	// Just sent message to user for let them know we received these order
+		      	$prepare_confirm_message = Omise_FBBot_Conversation_Generator::prepare_confirm_order_message( $order->get_order_number() );
+		      	$response = Omise_FBBot_HTTPService::send_message_to( $params['messenger_id'], $prepare_confirm_message );
+
+		      	// If merchant enable 3ds mode
+		      	if ( isset ( $charge['authorize_uri'] ) ) {
+		        	if ( wp_redirect( $charge['authorize_uri'] ) ) {
+		          		error_log( 'redirect to -> '. $charge['authorize_uri'] );
+		          		exit;
+		        	}
+		        
+		        	return;
+		      	}
+
+		      	// If merchant disable 3ds mode : normal mode
+		      	$redirect_uri =  site_url() . '/complete-payment';
+		      	if ( wp_redirect( $redirect_uri ) ) {
+		          	exit;
+		      	}
+
+		    } catch (Exception $e) {
+		      	$error_message = str_replace(" ", "%20", $e->getMessage());
+
+		      	// Redirect to error page
+		      	$redirect_uri =  site_url() . '/pay-on-messenger-error/?error_message=' . $error_message;
+		      	if ( wp_redirect( $redirect_uri ) ) {
+		          	exit;
+		      	}
+		    }
+		}
+
+  		private function is_omise_payment_page() {
+			global $wp;
+
+			return ( strtolower( $wp->request ) == $this->payment_page_url );
+		}
+
+		private function is_omise_payment_error_page() {
+			global $wp;
+			
+			return ( strtolower( $wp->request ) == $this->payment_error_url );
+		}
+
+		private function is_omise_purchase_complete_page() {
+			global $wp;
+			
+			return ( strtolower( $wp->request ) == $this->payment_purchase_complete_url );
+		}
+
+		private function is_accessible() {
+			global $wp;
+
+			return ( strtolower( $wp->request ) == $this->payment_page_url || strtolower( $wp->request ) == $this->payment_error_url );
```

Comment:
@igroomgrim I meant, we don't need to do `strtolower( $wp->request ) == $this->payment_page_url` because we already have `is_omise_payment_page()`. Can we use that one instead?